### PR TITLE
CASSANDRA-14667 Upgrade Dropwizard Metrics to 4.2.19

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: .build/parent-pom-template.xml [hunk 1]
```diff
@@ -628,17 +629,17 @@
...
         <groupId>io.dropwizard.metrics</groupId>
         <artifactId>metrics-jvm</artifactId>
-        <version>3.1.5</version>
+        <version>${metrics.version}</version>
       </dependency>
       <dependency>
         <groupId>io.dropwizard.metrics</groupId>
         <artifactId>metrics-logback</artifactId>
-        <version>3.1.5</version>
+        <version>${metrics.version}</version>
```
Comment (michaelsembwever) [inline .build/parent-pom-template.xml]: opinion: not a huge fan of properties like this when the use is all in one adjacent place.

Comment (Mmuzaf) [reply .build/parent-pom-template.xml]: I did this in accordance with the documentation:
https://metrics.dropwizard.io/4.2.0/getting-started.html

> Note
> Make sure you have a metrics.version property declared in your POM with the current version, which is 4.2.0.

Comment (michaelsembwever) [reply .build/parent-pom-template.xml]: I'm not sure they know what they are talking about.

What does 
```
grep -r  "metrics.version" ~/.m2/repository/io/dropwizard/metrics
```
give you?

Educate me please if I'm missing something, I see nothing in their parent pom, and we're not pulling in a bom.

Comment (Mmuzaf) [reply .build/parent-pom-template.xml]: I don't know the reasoning covered behind the message, but I understand it as follows: if a user is using the cassandra parent and wants to use additional modules provided by the dropwizard metrics project he/she should refer to the {metrics.version} inherited from the cassandra-parent when adding these dependencies. So if the cassandra project updates the {metrics.version} version of the dropwizard metrics library and a user wants to update the cassandra-parent accordingly, all dropwizard metrics extensions will be the same version and there will be no conflict between them.

But maybe I'm missing something, so I don't have anything against using the `4.2.19` directly.

Comment (michaelsembwever) [reply .build/parent-pom-template.xml]: transitive dependencies in maven resolve this for us.

there are edge cases, like bom files where properties are contextual, but i don't see that happening here…

Comment (Mmuzaf) [reply .build/parent-pom-template.xml]: Fixed :-)

**Pull request closed without merging**